### DISCUSSION TITLE
Unstable hovering , using flowdec

### DISCUSSION THREAD

**Discussion by iNavin** [bitcraze/discussions#966] [Q&A] [answered]
Hi,
i am using the crazyFlie 2.1 with flowdeck , using the Cfclient setup i am trying to hover the drone using the command based flight control - but the drone is not able hover , it is drifting in some other directions. I was following the lecture from - Crazyflie 101 EPFL-LIS lecture (2023) (youtube) , when the instructor has flow deck she was able to hover the drone, is there some suggestion on how to setup the drone - so that it could hover using the cflient

**Comment by knmcguire** [comment, marked as answer]
Hi! I'm actually the instructor :) Let me help you with that.

Please check if you have the following conditions:
- You should fly over a texture-rich, non-black surface. 
- There should be enough lighting in the room. Enough such that there is no shadow directly under the crazyflie (it might track that and drift)
- Check for any error messages for the flowdeck in the [console tab.](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/console_tab/)
- Check the [flow logging variables(motion.x and y)](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#motion) in the [plotter tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/plotter_tab/) (also a good way to check if the surface you are flying over is adequate.)

**Reply by iNavin** [reply]
thanks your suggestion worked.

**Reply by knmcguire** [reply]
Great! Is there any specific tip that worked the best?

If you are experiencing any other issue than this, start a new discussion.
